lib/pure/strmisc.nim

```diff
@@ -85,6 +86,139 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+
+template parseFloatThousandSepImpl(s: var string; sep: static char; decimalDot: static char): float =
+  assert sep notin {'-', ' '} and decimalDot notin {'-', ' '} and sep != decimalDot
+
+  template bail(m: string) =
+    raise newException(ValueError, "Invalid float containing thousand separators, " & m)
+
+  if likely(s.len > 1): # Allow "0" thats valid, is 0.0
+    var idx, successive: int
+    var afterDot, lastWasDot, lastWasSep, hasAnySep, isNegative: bool
+    while idx < s.len:
+      case s[idx]
+      of '0' .. '9':  # Digits
+        if hasAnySep and successive > 2:
+          bail("more than 3 digits between thousand separators.")
+        else:
+          lastWasSep = false
+          lastWasDot = false
+          inc successive
+          inc idx
+      of sep:  # Thousands separator
+        if unlikely(isNegative and idx == 1 or idx == 0):
+          bail("string starts with thousand separator.")
+        elif lastWasSep:
+          bail("two separators in a row.")
+        elif afterDot:
+          bail("separator found after decimal dot.")
+        else:
+          s.delete(idx, idx)
```

Comment:
Instead of ``s.delete`` copy ``s`` into a sanitized buffer like so:

```nim

  var buf = newStringOfCap(s.len)
  var i = 0
  while i < s.len:
     ...
     buf.add s[i]
  ...
  parseFloat(buf)
```